### PR TITLE
Fix create variable code action test fail, when the diagnostic codes are changed

### LINKED ISSUE #38883: [Bug]: Create variable code action test fail when the diagnostic codes are changed
### Description

Create variable code action test fail when the diagnostic codes are changed

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Other Area

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes #38883

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java [hunk 1]
```diff
@@ -71,11 +71,12 @@
 public class CreateVariableWithTypeCodeAction extends CreateVariableCodeAction {
 
     private static final String NAME = "Create variable with type";
+    private static final String DIAGNOSTIC_MESSAGE_KEY = "cannot.infer.typedesc.argument.without.cet";
 
     @Override
     public boolean validate(Diagnostic diagnostic, DiagBasedPositionDetails positionDetails,
                             CodeActionContext context) {
-        return diagnostic.diagnosticInfo().code().equals("BCE4046")
+        return diagnostic.diagnosticInfo().messageFormat().equals(DIAGNOSTIC_MESSAGE_KEY)
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java:79]: Is using the `message key` more reliable than using diagnostic codes? Ideally diagnostic codes aren't supposed to change frequently. If a diagnostic code changes (or new diagnostic codes are added), adding/modifying the diagnostic code here is the best approach IMO.

**Review by IMS94 — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java (hunk 1 above)
Comment (mohanvive) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java:79]: Yes, +1. We are gradually moving away from diagnostic messages and using codes in code actions. Hence, let's change the fix accordingly please.

Comment (SasinduDilshara) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java:79]: In this scenario, due to the revert of client declaration related changes diagnostic codes are changed.

Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java:79]: Yes, other thing is, there can be 2 diagnostics with the same message, but with 2 different diagnostic codes and different properties. We rely on diag properties to determine some contextual information.

Comment (SasinduDilshara) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java]: Update the changes.

**Review by IMS94 — APPROVED** [review]

**Pull request merged**